x/staking/keeper/slash.go

```diff
@@ -145,34 +145,42 @@ func (k Keeper) Unjail(ctx sdk.Context, consAddr sdk.ConsAddress) {
 // (the amount actually slashed may be less if there's
 // insufficient stake remaining)
 func (k Keeper) slashUnbondingDelegation(ctx sdk.Context, unbondingDelegation types.UnbondingDelegation,
-	infractionHeight int64, slashFactor sdk.Dec) (slashAmount sdk.Int) {
+	infractionHeight int64, slashFactor sdk.Dec) (totalSlashAmount sdk.Int) {
 
 	now := ctx.BlockHeader().Time
+	totalSlashAmount = sdk.ZeroInt()
 
-	// If unbonding started before this height, stake didn't contribute to infraction
-	if unbondingDelegation.CreationHeight < infractionHeight {
-		return sdk.ZeroInt()
-	}
+	// perform slashing on all entries within the unbonding delegation
+	for i, entry := range unbondingDelegation.Entries {
 
-	if unbondingDelegation.MinTime.Before(now) {
-		// Unbonding delegation no longer eligible for slashing, skip it
-		// TODO Settle and delete it automatically?
-		return sdk.ZeroInt()
-	}
+		// If unbonding started before this height, stake didn't contribute to infraction
+		if entry.CreationHeight < infractionHeight {
+			continue
+		}
 
-	// Calculate slash amount proportional to stake contributing to infraction
-	slashAmountDec := slashFactor.MulInt(unbondingDelegation.InitialBalance.Amount)
-	slashAmount = slashAmountDec.TruncateInt()
+		if entry.IsMature(now) {
+			// Unbonding delegation no longer eligible for slashing, skip it
+			// TODO Settle and delete it automatically?
```

Comment:
Is this vital? What does this really mean?

Reply:
```suggestion
```

Reply:
no should be remove was just copied